include/grpc++/impl/codegen/async_stream.h

```diff
@@ -225,7 +275,8 @@ class ClientAsyncWriter final : public ClientAsyncWriterInterface<W> {
   Call call_;
   CallOpSet<CallOpSendInitialMetadata> init_ops_;
   CallOpSet<CallOpRecvInitialMetadata> meta_ops_;
-  CallOpSet<CallOpSendMessage> write_ops_;
+  CallOpSet<CallOpSendInitialMetadata, CallOpSendMessage, CallOpClientSendClose>
```

Comment:
Agreed with Craig. You can only have one of init_, writes_done_, or write_ outstanding at a time since they all include write-side operations so this should save a few entries.